packages/plugin-core/src/commands/AddExistingVaultCommand.ts

```diff
@@ -0,0 +1,504 @@
+
+enum VaultType {
+  LOCAL = "local",
+  REMOTE = "remote",
+}
+
+export class AddExistingVaultCommand extends BasicCommand<
+  CommandOpts,
+  CommandOutput
+> {
+  key = DENDRON_COMMANDS.ADD_EXISTING_VAULT.key;
+
+  constructor(private _ext: IDendronExtension) {
+    super();
+  }
+
+  generateRemoteEntries = (): SourceQuickPickEntry[] => {
+    return DENDRON_REMOTE_VAULTS.map(
+      ({ name: label, description, data: src }): SourceQuickPickEntry => {
+        return { label, description, src };
+      }
+    ).concat([
+      {
+        label: "custom",
+        description: "custom endpoint",
+        alwaysShow: true,
+        src: "",
+      },
+    ]);
+  };
+
+  /** A regular, non-self contained vault. */
+  async gatherVaultStandard(
+    sourceType: VaultRemoteSource,
+    vaultDestination: string
+  ): Promise<CommandOpts | undefined> {
+    let sourceName: string | undefined;
+    if (sourceType === VaultType.REMOTE) {
+      // eslint-disable-next-line  no-async-promise-executor
+      const out = new Promise<CommandOpts | undefined>(async (resolve) => {
+        const qp = VSCodeUtils.createQuickPick<SourceQuickPickEntry>();
+        qp.ignoreFocusOut = true;
+        qp.placeholder = "choose a preset or enter a custom git endpoint";
+        qp.items = this.generateRemoteEntries();
+        qp.onDidAccept(async () => {
+          const value = qp.value;
+          const selected = qp.selectedItems[0];
+          if (selected.label === "custom") {
+            if (PickerUtilsV2.isInputEmpty(value)) {
+              return window.showInformationMessage("please enter an endpoint");
+            }
+            selected.src = qp.value;
+          }
+          const sourceRemotePath = selected.src;
+          const path2Vault =
+            selected.label === "custom"
+              ? GitUtils.getRepoNameFromURL(sourceRemotePath)
+              : selected.label;
+          const placeHolder = path2Vault;
+          sourceName = await VSCodeUtils.showInputBox({
+            prompt: "Name of new vault (optional, press enter to skip)",
+            value: placeHolder,
+          });
+          qp.hide();
+          return resolve({
+            type: sourceType!,
+            name: sourceName,
+            path: vaultDestination,
+            pathRemote: sourceRemotePath,
+          });
+        });
+        qp.show();
+      });
+      return out;
+    }
+    const placeHolder = path.basename(vaultDestination);
+    sourceName = await VSCodeUtils.showInputBox({
+      prompt: "Name of new vault (optional, press enter to skip)",
+      placeHolder,
+    });
+    return {
+      type: sourceType,
+      name: sourceName,
+      path: vaultDestination,
+    };
+  }
+
+  async gatherDestinationFolder() {
+    const defaultUri = Uri.file(this._ext.getDWorkspace().wsRoot);
+    // Prompt user where to create new vault
+    const options: OpenDialogOptions = {
+      canSelectMany: false,
+      openLabel: "Select vault to add",
```

Comment:
Open an existing folder of markdown files

Reply:
opens the workspace root by default